### ISSUE TITLE
duplicate resids hide residues in atom.residues

### ISSUE THREAD

**Issue by orbeckst** [MDAnalysis/mdanalysis#984]
I used the PDB from https://github.com/MDAnalysis/mdanalysis/issues/975#issuecomment-246074022 `asdf.pdb` which has no chain ID but duplicated resids.
### Expected behaviour

All residues should be separately resolved in `atoms.residues`.
### Actual behaviour

Instead, only one residue for each resid is visible.

All atoms are present.
### Code to reproduce the behaviour

Download [asdf.py](https://dl.dropboxusercontent.com/u/49250897/asdf.pdb) from dropbox.

``` python
import MDAnalysis as mda
u = mda.Universe("asdf.pdb")
print(len(u.atoms))
#29879
print(len(u.residues))
431
# the correct number of residues:
print(u.atoms.CA.n_atoms)
#1849
```
### Currently version of MDAnalysis:

(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.15.0, 0.15.1-dev

**Comment by richardjgowers** [comment]
This is a little confusing because we've also had an issue for unordered atoms at #193 where we wanted to consolidate the duplicate resids into a single residue

**Comment by orbeckst** [comment]
You are right, the desired fix in this issue and the solution to #193 are mutually exclusive. I would argue that #193 is the exception and this one is the more common case. Do we even take segid into account in order to build different residues, i.e., is there currently *any* way that we can chemically correctly build separate  residues even if they got the same label in the file? 

IMO, what should happen is this:
* MDAnalysis assigns a *resid* (or should we call it *resindex*), which is unique – how is this handled in #363?
* Set the *resnum* to whatever is in the file and treat it as a label.

There's still a question as to how to deal with files where the ordering of residues is random (#193). I think in this case it would be better to have the user set some flag to indicate this special reading mode instead of being too smart. (I don't even want to consider a distance search for atoms belonging to a residue – that just opens another can of worms...).

**Comment by richardjgowers** [comment]
@orbeckst so currently resids/segids are being used as keys and kept as in the topology file, and MDAnalyis is inventing a resindex which is how it keeps track of residues.

(Disclaimer, I don't use PDB for their actual purpose of bio things)

So the desired behaviour for the PBD parser is that if a resid changes, it's considered a new residue even if it's an existing key?  This then makes the `Resid` property relegated to something more like resname where there can be repeated values.

**Comment by orbeckst** [comment]
> so currently resids/segids are being used as keys and kept as in the topology file, and MDAnalyis is inventing a resindex which is how it keeps track of residues.

Ok, that also answers my questions in https://github.com/MDAnalysis/mdanalysis/issues/985#issuecomment-246839339

> So the desired behaviour for the PBD parser is that if a resid changes, it's considered a new residue even if it's an existing key? 

I think that this makes more sense as the default behavior than what was implemented for #193. In principle we could check that the residues are on different chains but even that is not always true as in the example. The question is really what assumption is the safer one to make: separate residues or joined residues.

I now argue that joined residues is less common and if anything, violates the PDB standard more than assuming that separate resids describe separate residues.

Admittedly, proper PDB files have the [SEQRES](http://www.wwpdb.org/documentation/file-format-content/format33/sect3.html#SEQRES) section that details exactly the separate chemical entities (and the MMTF format #913 also has this information built in). PDB says (implicitly in the SEQRES record) that different chains (*chainID*) may contain different residues with the same *resSeq* number in the [ATOM](http://www.wwpdb.org/documentation/file-format-content/format33/sect9.html#ATOM) section. The ATOM section also states clearly:

**"The sequence implied by ATOM records must be identical to that given in SEQRES [...]"**

> This then makes the Resid property relegated to something more like resname where there can be repeated values.

Or rather *resnum*.
